src/checkpoint/checkpoint.ts

```diff
@@ -70,32 +63,86 @@ export default class Checkpoint {
 
   public async start() {
     this.log.debug('starting');
+
+    // TODO(perfectmak): Validate config before starting
+
+    await this.store.createStore();
+
     const blockNum = await this.getStartBlockNum();
     return await this.next(blockNum);
   }
 
   public async reset() {
     this.log.debug('reset');
+
+    await this.store.deleteStore();
+    await this.store.createStore();
+
     await this.entityController.createEntityStores(this.mysql);
   }
 
+  /**
+   * Registers the blocks to use as a skip list for checkpoint while
+   * indexing relevant blocks. Using this seed function can significantly
+   * reduce the time for Checkpoint to re-index blocks.
+   *
+   * This should be called before the start() method is called.
+   *
+   */
+  public async seedCheckpoint(blocks: number[]): Promise<void> {
+    await this.store.createStore();
+
+    const checkpoints: CheckpointRecord[] = [];
+
+    this.sourceContracts.forEach(contractAddress => {
```

Comment:
This assume each contracts from the config got a checkpoint at specific blocks, but it wont necessary be the case, we would endup with incorrect data saying contract A have an event on block X but doesn't.

Reply:
Hmmm... I thought about this as well, I went with this approach because I assumed it will be easier to use the list of contracts in the `config` object provided and just ask the user to provide blocks where at least one of those contracts has an event. But from a data integrity perspective, this could be faulty and will prevent sharing blocks with other checkpoint instances.

The ideal solution will be to have this method take an array of blockNumber <--> contractAddress pairs (or an object representing that structure). And I'm fine with that as well. I will make the change.

Reply:
This is has been fixed here: https://github.com/snapshot-labs/sx-api/pull/39/commits/4529205a23e6f2a990f5f7858c384f8b3ea74247